### PR TITLE
MTA-1309: Do not allow input and output dirs to be the same for transform rules

### PR DESCRIPTION
**Pull request by eemcmullan**
https://issues.redhat.com/browse/MTA-1309

### CODE REVIEW

File: cmd/shimconvert.go [hunk 1]
```diff
@@ -77,6 +77,11 @@ func (w *windupShimCommand) Validate() error {
 	if w.input == nil || len(w.input) == 0 {
 		return fmt.Errorf("input for rule file or directory must not be empty")
 	}
+	for _, r := range w.input {
+		if r == w.output {
+			return fmt.Errorf("input rule directory and output directory must be different")
```
Comment (abrugaro) [inline cmd/shimconvert.go:82]: **question:** Will this error be triggered also if the paths are different but just by slashes?

Example:
* input: /path/to/folder/ (Slash at the end)
* output: /path/to/folder (No slash at the end)

Or

* input: /path\to\folder (backslashes)
* output: /path/to/folder

I'm new to Go so I don't know if the paths are being sanitized before this condition.

Thanks!

**Comment by pranavgaikwad** [comment]
@eemcmullan need to update line numbers in demo-output because of https://github.com/konveyor/analyzer-lsp/commit/dca2d151b334da9b55e77ae88f4b49671989773a

File: cmd/shimconvert.go (hunk 1 above)
Comment (pranavgaikwad) [reply cmd/shimconvert.go:82]: @abrugaro great point, paths indeed need to be cleaned before comparing, thank you

**Review by pranavgaikwad — CHANGES_REQUESTED** [review]
Clean paths, and increment line numbers by 1 in demo-output for https://github.com/konveyor/kantra/blob/9e3f53996c9e2da850b963786420c75813984d13/test-data/analysis-output.yaml#L1440 and https://github.com/konveyor/kantra/blob/9e3f53996c9e2da850b963786420c75813984d13/test-data/analysis-output.yaml#L1472

**Review by pranavgaikwad — APPROVED** [review]
LGTM just need DCO

**Pull request merged**